BDArmory/UI/BDAEditorArmorWindow.cs

```diff
@@ -430,52 +454,34 @@ void WindowArmor(int windowID)
             if (!BDArmorySettings.RESET_HULL)
             {
                 line += 0.5f;
-                showHullMenu = GUI.Toggle(new Rect(10, (line + armorLines + StatLines) * lineHeight, 280, lineHeight),
-                    showHullMenu, StringUtils.Localize("#LOC_BDArmory_Armor_HullMat"), showHullMenu ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button);
-                HullLines += 1.15f;
-
-                if (showHullMenu)
+                if (!hullslist)
                 {
-                    if (isSteel != (isSteel = GUI.Toggle(new Rect(10, (line + armorLines + StatLines + HullLines) * lineHeight, 280, lineHeight), isSteel, StringUtils.Localize("#LOC_BDArmory_Steel"), isSteel ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button)))
-                    {
-                        if (isSteel)
-                        {
-                            isWood = false;
-                            isAluminium = false;
-                            hullmat = 3;
-                            CalculateArmorMass(true);
-                        }
-                    }
-                    HullLines += 1.15f;
-                    if (isWood != (isWood = GUI.Toggle(new Rect(10, (line + armorLines + StatLines + HullLines) * lineHeight, 280, lineHeight), isWood, StringUtils.Localize("#LOC_BDArmory_Wood"), isWood ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button)))
-                    {
-                        if (isWood)
-                        {
-                            isAluminium = false;
-                            isSteel = false;
-                            hullmat = 1;
-                            CalculateArmorMass(true);
-                        }
-                    }
-                    HullLines += 1.15f;
-                    if (isAluminium != (isAluminium = GUI.Toggle(new Rect(10, (line + armorLines + StatLines + HullLines) * lineHeight, 280, lineHeight), isAluminium, StringUtils.Localize("#LOC_BDArmory_Aluminium"), isAluminium ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button)))
-                    {
-                        if (isAluminium)
-                        {
-                            isWood = false;
-                            isSteel = false;
-                            hullmat = 2;
-                            CalculateArmorMass(true);
-                        }
-                    }
-                    HullLines += 1.15f;
-                    if (!isSteel && !isWood && !isAluminium)
+                    FillHullList();
+                    hullslist = true;
+                }
+                GUIStyle listStyle = new GUIStyle(BDArmorySetup.BDGuiSkin.button);
```

Comment:
The `listStyle` definitions should be done only once. You can put 
```csharp
GUIStyle listStyle = new GUIStyle(BDArmorySetup.BDGuiSkin.button);
```
as one of the `BDAEditorArmorWindow` fields and set
```csharp
listStyle.fixedHeight = 18; //make list contents slightly smaller
```
in `Start`.